### PR TITLE
feat(mailer): support file attachments in sendMail()

### LINKED ISSUE #3374: feat(mailer): support file attachments in sendMail()
## Problem

Data from scrap runs is currently injected inline in the email HTML body. Users want it as a proper file attachment (CSV or JSON).

## Proposed API

\`\`\`js
sendMail({
  to, subject, template, params,
  attachments: [{ filename: 'data.csv', content: 'col1,col2\nval1,val2' }]
})
\`\`\`

## Provider changes

**Resend** (`provider.resend.js`):
\`\`\`js
attachments: mail.attachments?.map(a => ({
  filename: a.filename,
  content: Buffer.from(a.content).toString('base64'),
}))
\`\`\`

**Nodemailer** (`provider.nodemailer.js`):
\`\`\`js
attachments: mail.attachments
\`\`\`

## Context

Needed for `emailNotif` post-run emails in downstream projects (trawl, comes, etc.).

### PR DESCRIPTION
**Pull request by PierreBrisorgueil**
## Summary

- Add optional `attachments` array parameter to `sendMail()`
- **Resend provider**: encodes attachment content to base64 before sending
- **Nodemailer provider**: passes attachments through directly (nodemailer native format)
- JSDoc updated on all three files; unit tests added for both providers

## API

```js
sendMail({
  to, subject, template, params,
  attachments: [{ filename: 'data.csv', content: 'col1,col2\nval1,val2' }]
})
```

## Test plan

- [x] `provider.resend.unit.tests.js` — new test: sends attachment base64-encoded
- [x] `mailer.unit.tests.js` — new test: attachments forwarded to provider
- [x] All 416 unit tests passing, lint clean

Closes #3374

## Summary by CodeRabbit

* **New Features**
  * Added support for email attachments with Buffer content, enabling more flexible attachment handling beyond string-based content.

* **Tests**
  * Added comprehensive unit tests for email provider attachment functionality, validating proper handling of array and buffer-based attachments.

### CODE REVIEW

File: lib/helpers/mailer/provider.nodemailer.js [hunk 1]
```diff
@@ -5,7 +5,9 @@ export default class NodemailerProvider {
     this.transport = nodemailer.createTransport(options);
   }
 
-  async send({ from, to, subject, html }) {
-    return this.transport.sendMail({ from, to, subject, html });
+  async send({ from, to, subject, html, attachments }) {
+    const payload = { from, to, subject, html };
+    if (attachments?.length) payload.attachments = attachments;
+    return this.transport.sendMail(payload);
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply lib/helpers/mailer/provider.nodemailer.js]: Fixed in c4ae8e17: added `provider.nodemailer.unit.tests.js` covering: send without attachments, send with string attachments, send with Buffer attachments, and the empty-attachments-array edge case (verifies no `attachments` key in payload).

File: lib/helpers/mailer/index.js [hunk 2]
```diff
@@ -44,6 +44,9 @@ const isConfigured = () => !!(config.mailer && config.mailer.from && !String(con
  * @param {string} mail.subject - Email subject
  * @param {string} mail.template - Template name (without .html)
  * @param {Object} mail.params - Template parameters
+ * @param {Array} [mail.attachments] - Optional attachments array
+ * @param {string} mail.attachments[].filename - Attachment filename
+ * @param {string} mail.attachments[].content - Attachment content (string or Buffer)
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply lib/helpers/mailer/index.js]: Fixed in c4ae8e17: updated the type annotation to `string|Buffer` in ResendProvider JSDoc to accurately reflect accepted inputs.
